src/Client/Updater.php

```diff
@@ -472,14 +475,14 @@ private function updateRoot(RootMetadata &$rootData): void
     }
 
     /**
-     * Gets the current time.
+     * Gets the time beyond which any new metadata will be considered expired.
      *
      * @return \DateTimeImmutable
-     *    The current time.
+     *    A time beyond which any new metadata will be considered expired.
      * @throws \Tuf\Exception\FormatException
      *    Thrown if time format is not valid.
      */
-    private function getCurrentTime(): \DateTimeImmutable
+    protected function getMetadataExpirationTime(): \DateTimeImmutable
```

Comment:
I don't think we should have this as a function that would call multiple times and it retrieve the time(even if it is fake) each time.

What if we introduced a `protected` `setMetadataExpiration()` that would be called at the very start of `refresh` this would set an a `private $expirationTime`(we could copy our existing logic for getting `$now`.

And then we have a `private` `getMetadataExpirationTime()` that just returns `private $expirationTime` and throws an \`LogicException` if it is called before `setMetadataExpiration()`

That way classes could override `setMetadataExpiration()` but not `getMetadataExpirationTime()` so it would be easy to change how logic it should be before expiration but they couldn't mess with the basic logic.

Reply:
I implemented this but had to make everything  protected for it to overridable.

Reply:
Makes sense; I'm wondering, however, if we really need `getMetadataExpirationTime` to be protected. That might make more sense as a private method. We want to allow subclasses set _set_ an arbitrary expiration time, but once it's set, shouldn't we treat it as an internal aspect of TUF's workings?

Reply:
sounds good. created `createExpirationDate()` as protected and made everything else private